cmd/flux/diff_artifact.go

```diff
@@ -103,10 +111,112 @@ func diffArtifactCmdRun(cmd *cobra.Command, args []string) error {
 		}
 	}
 
-	if err := ociClient.Diff(ctx, url, diffArtifactArgs.path, diffArtifactArgs.ignorePaths); err != nil {
+	ociContents, err := fetchContentsFromURL(ctx, url)
+	if err != nil {
 		return err
 	}
 
-	logger.Successf("no changes detected")
+	compareToContents, err := fetchContentsFromLocalPath(diffArtifactArgs.path)
+	if err != nil {
+		return err
+	}
+
+	diffOutput, err := diffContents(ociContents, compareToContents)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Diff between OCI artifact (%s) and local path (%s):\n", ociURL, diffArtifactArgs.path)
+
+	// Show verbose output if the --verbose flag is set
+	if verbose, _ := cmd.Flags().GetBool("verbose"); verbose {
+		fmt.Println("Printing diff...")
+	}
+
+	fmt.Println(diffOutput)
+
 	return nil
 }
+
+func fetchContentsFromURL(ctx context.Context, url string) ([]byte, error) {
+	img, err := crane.Pull(url)
+	if err != nil {
+		return nil, fmt.Errorf("failed to pull OCI artifact: %w", err)
+	}
+
+	layers, err := img.Layers()
+	if err != nil {
+		return nil, fmt.Errorf("failed to list layers: %w", err)
+	}
+
+	if len(layers) < 1 {
+		return nil, fmt.Errorf("no layers found in OCI artifact")
+	}
+
+	// Create a buffer to store the contents of the artifact
+	var contentsBuffer bytes.Buffer
+
+	// Extract the content of the first layer
+	layerContent, err := layers[0].Compressed()
+	if err != nil {
+		return nil, fmt.Errorf("failed to extract layer content: %w", err)
+	}
+	defer layerContent.Close()
+
+	// Use tar.NewReader to read the contents from the OCI artifact
+	tarReader := tar.NewReader(layerContent)
+	for {
+		_, err := tarReader.Next()
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return nil, fmt.Errorf("failed to read tar header: %w", err)
+		}
+
+		// Read the file contents and write them to the buffer
+		if _, err := io.Copy(&contentsBuffer, tarReader); err != nil {
+			return nil, fmt.Errorf("failed to read tar entry contents: %w", err)
+		}
+	}
```

Comment:
Instead of this looking at just file contents, I think it would be much better if path data would be taken into account (which also can be done for local directories, as they are expected to have the same structure as a decompressed tar).

This way, not only differences in file content would be observed. But also things being moved (or swapped) from file A -> B, files changing from path, etc.